### PR TITLE
Bugfix-42: Create proper pagination handlers for list views.

### LINKED ISSUE #42: Fix paging in volunteers and requests views
Jump to last page doesn't work well, maybe allow jump to specific page (long page list)

### PR DESCRIPTION
**Pull request by astar26**
Resolves #42

### CODE REVIEW

File: server/templates/server/help_table.html [hunk 1]
```diff
@@ -9,6 +9,22 @@
...
+        {% for i in pages_before %}
+            <a href="{% url 'show_all_help_request' %}/{{ i }}?{{ request.META.QUERY_STRING }}">{{ i }}</a>
+        {% endfor %}
+        &nbsp;עמוד {{ page }} מתוך {{ num_pages }}&nbsp;
+        {% for i in pages_after %}
+            <a href="{% url 'show_all_help_request' %}/{{ i }}?{{ request.META.QUERY_STRING }}">{{ i }}</a>
+        {% endfor %}
+        {% if num_pages not in pages_after and num_pages != page %}
+            <a href="{% url 'show_all_help_request' %}/{{ num_pages }}?{{ request.META.QUERY_STRING }}">&gt;&gt;</a>
+        {% endif %}
```
Comment (itay-sho) [inline server/templates/server/help_table.html]: fix indentation

Comment (itay-sho) [inline server/templates/server/help_table.html]: you are using this piece of code on multiple places. create a template for that and read about {% include %}

File: server/views.py [hunk 2]
```diff
@@ -179,8 +180,11 @@ def show_all_volunteers(request, page=1):
 
     final_data = paginator.page(page)
 
+    list_pages_before = range(max(1, page - PAGINATION_SHORTCUT_NUMBER), page)
+    list_pages_after = range(page + 1, min(page + PAGINATION_SHORTCUT_NUMBER + 1, paginator.num_pages + 1))
```
Comment (itay-sho) [inline server/views.py]: readability is very limited here

File: server/views.py [hunk 3]
```diff
@@ -179,8 +180,11 @@ def show_all_volunteers(request, page=1):
...
+
     context = {'volunteer_data': final_data, 'availability_now_id': availability_now_id, 'page': page,
-               'num_pages': paginator.num_pages}
+               'num_pages': paginator.num_pages, 'pages_before': list_pages_before, 'pages_after': list_pages_after}
```
Comment (itay-sho) [inline server/views.py]: indentation.
do it like:
context = {
    'foo': 'bar',
    'foo2': 'bar2',
    ....
}

Comment (itay-sho) [reply server/views.py]: well github is removing my spaces, but the fields should be one tab inside

File: server/views.py [hunk 4]
```diff
@@ -233,7 +237,11 @@ def show_all_help_request(request, page=1):
     paginator = Paginator(match_qs, RESULTS_IN_PAGE)
     match_qs = paginator.page(page)
 
-    context = {'help_requests': match_qs, 'page': page, 'num_pages': paginator.num_pages}
+    list_pages_before = range(max(1, page - PAGINATION_SHORTCUT_NUMBER), page)
+    list_pages_after = range(page + 1, min(page + PAGINATION_SHORTCUT_NUMBER + 1, paginator.num_pages + 1))
```
Comment (itay-sho) [inline server/views.py]: duplication. consider exporting into a function

File: server/templates/server/help_table.html (hunk 1 above)
Comment (astar26) [reply server/templates/server/help_table.html]: Fixed

Comment (astar26) [reply server/templates/server/help_table.html]: Moved for now to 2 snippets, will be moved again in the future.

File: server/views.py (hunk 2 above)
Comment (astar26) [reply server/views.py]: Added docs

File: server/views.py (hunk 3 above)
Comment (astar26) [reply server/views.py]: Done

File: server/templates/server/help_table.html (hunk 1 above)
Comment (itay-sho) [reply server/templates/server/help_table.html]: don't resolve conversations - just write fix and the person who are doing the CR will press resolve

File: server/views.py [hunk 5]
```diff
@@ -64,6 +65,19 @@ def index(request):
     return render(request, 'server/server_index.html', context)
 
 
+def get_close_pages(page, max_page):
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
def get_close_pages(current_page, pages_count):
```

File: server/views.py [hunk 6]
```diff
@@ -64,6 +65,19 @@ def index(request):
...
+    """
+    Generates lists of pages to link to, before and after, while considering page number and current location in mind.
+    Ammount of pages to each side is defined in PAGINATION_SHORTCUT_NUMBER.
+    :param page: current page number
+    :param max_page: last page number
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
    :param current_page: current page number
    :param pages_count: total amount of pages
```

File: server/views.py [hunk 7]
```diff
@@ -64,6 +65,19 @@ def index(request):
...
+    :return: (list of pages to link to before current page, list of pages to link to after current page)
+    """
+    list_pages_before = range(max(1, page - PAGINATION_SHORTCUT_NUMBER), page)
+    list_pages_after = range(page + 1, min(page + PAGINATION_SHORTCUT_NUMBER + 1, max_page)+1)
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
    list_pages_before = range(max(1, current_page - PAGINATION_SHORTCUT_NUMBER), current_page)
    list_pages_after = range(current_page + 1, min(current_page + PAGINATION_SHORTCUT_NUMBER + 1, pages_count)+1)
```

File: server/views.py [hunk 8]
```diff
@@ -176,11 +190,15 @@ def show_all_volunteers(request, page=1):
...
-    context = {'volunteer_data': final_data, 'availability_now_id': availability_now_id, 'page': page,
-               'num_pages': paginator.num_pages}
+    list_pages_before, list_pages_after = get_close_pages(page, paginator.num_pages)
+
+    context = {'volunteer_data': final_data,
+               'availability_now_id': availability_now_id,
+               'page': page,
+               'num_pages': paginator.num_pages,
+               'pages_before': list_pages_before,
+               'pages_after': list_pages_after}
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
    context = {
        'volunteer_data': final_data,
        'availability_now_id': availability_now_id,
        'page': page,
        'num_pages': paginator.num_pages,
        'pages_before': list_pages_before,
        'pages_after': list_pages_after
    }
```

File: server/views.py [hunk 9]
```diff
@@ -233,7 +251,13 @@ def show_all_help_request(request, page=1):
...
     match_qs = paginator.page(page)
 
-    context = {'help_requests': match_qs, 'page': page, 'num_pages': paginator.num_pages}
+    list_pages_before, list_pages_after = get_close_pages(page, paginator.num_pages)
+
+    context = {'help_requests': match_qs,
+               'page': page,
+               'num_pages': paginator.num_pages,
+               'pages_before': list_pages_before,
+               'pages_after': list_pages_after}
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
    context = {
        'help_requests': match_qs,
        'page': page,
        'num_pages': paginator.num_pages,
        'pages_before': list_pages_before,
        'pages_after': list_pages_after
    }
```

File: server/templates/server/help_table.html (hunk 1 above)
Comment (itay-sho) [reply server/templates/server/help_table.html]: I'm opening issue for that ^

**Pull request merged**